# **PROJECT TITLE: EVALUATING CLIENTS CREDITWORTHINESS**

![Bank](Images/Banks.png)

# **PROJECT OVERVIEW**

# **OBJECTIVES**

In [1]:
# Importing libraries
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# **1. DATA LOADING & EXPLORATION**

In [2]:
# Loading the csvfile into a pandas DataFrame
df_Bank = pd.read_csv('Data/bank_credit_scoring.csv')
df_Bank

Задолженность  Просрочка, дни  Первоначльный лимит   BIRTHDATE  \
0            6063.50               3               7000.0  1983-07-08   
1            3765.04               0               5000.0  1987-12-19   
2            2067.66               0               2650.0  1966-02-28   
3            2370.39               0               3000.0  1972-02-18   
4            2280.55               0               3000.0  1997-02-23   
...              ...             ...                  ...         ...   
18415           0.00               0              10850.0  1999-11-08   
18416           0.00               0              13000.0  1993-06-03   
18417           0.00               0               3500.0  2002-11-05   
18418           0.00               0              10000.0  2003-09-15   
18419           0.00               0               1500.0  1994-01-14   

           SEX                  EDU   INCOME  TERM Рейтинг кредитной истории  \
0      Мужской  Среднее специальное   703.07    60                        A1   
1      Женский               Высшее  1693.68    60                        B2   
2      Женский              Среднее   724.49    60                        C2   
3      Женский  Среднее специальное  1045.84    60                        A3   
4      Женский               Высшее  1092.65    60                        B3   
...        ...                  ...      ...   ...                       ...   
18415  Мужской               Высшее  3563.75    60                        B1   
18416  Мужской  Среднее специальное  2865.72    36                       NaN   
18417  Мужской  Среднее специальное  3363.05    24                        C2   
18418  Мужской              Среднее  1670.13    60                        C1   
18419  Мужской               Высшее  3517.90    60                        C1   

                  LV_AREA LV_SETTLEMENTNAME                INDUSTRYNAME   PDN  \
0      Гомельская область         КОСАКОВКА                         АПК  0.98   
1                     NaN             МИНСК       Финансы и страхование  0.17   
2             ГРОДНЕНСКАЯ              ЁДКИ  Медицина и здравоохранение  0.40   
3                 МИНСКАЯ          ФАНИПОЛЬ  Государственное управление  0.29   
4                     NaN             Минск  Медицина и здравоохранение  0.63   
...                   ...               ...                         ...   ...   
18415            г. Минск             МИНСК   Информационные технологии  0.40   
18416     Минская область              Жалы               Строительство  0.19   
18417     Минская область         Солигорск               Строительство  0.19   
18418            г. Минск             МИНСК                    Торговля  0.28   
18419   Брестская область            КОБРИН                  МВД/МЧС/МО  0.26   

       CLIENTID  SCORINGMARK  UNDERAGECHILDRENCOUNT  VELCOMSCORING  \
0        919517        233.0                      0            NaN   
1        539353          NaN                      1            NaN   
2        818913          NaN                      0            NaN   
3        691598        216.0                      0            NaN   
4        425193          NaN                      0            NaN   
...         ...          ...                    ...            ...   
18415    687867        344.0                      0           -2.0   
18416    984627        108.0                      2           -2.0   
18417   1060373        129.0                      0           -2.0   
18418   1056128        179.0                      0           -2.0   
18419   1061453        269.0                      0           -2.0   

       FAMILYSTATUS  
0                 1  
1                 1  
2                 2  
3                 2  
4                 2  
...             ...  
18415             1  
18416             1  
18417             2  
18418             2  
18419             2  

[18420 rows x 18 columns]

Some columns names are in Russian.

### **TRANSLATING  RUSSIAN TO ENGLISH**

In [12]:
# Translate Column Names by Remaning them
df_Bank = df_Bank.rename(columns={
    'Задолженность': 'DEBT',
    'Просрочка, дни': 'OVERDUE_DAYS',
    'Первоначльный лимит': 'INITIAL_LIMIT',
    'Рейтинг кредитной истории': 'CREDIT_HISTORY_RATING',
    'LV_AREA': 'LIVING_AREA',
    'LV_SETTLEMENTNAME': 'SETTLEMENT_NAME',
    'INDUSTRYNAME': 'INDUSTRY_NAME',
    'PDN': 'PDN',
    'CLIENTID': 'CLIENT_ID',
    'SCORINGMARK': 'SCORING_MARK',
    'UNDERAGECHILDRENCOUNT': 'UNDERAGE_CHILDREN_COUNT',
    'VELCOMSCORING': 'VELCOM_SCORING',
    'FAMILYSTATUS': 'FAMILY_STATUS',
    'SEX': 'SEX',
    'EDU': 'EDU',
    'INCOME': 'INCOME',
    'TERM': 'TERM',
    'BIRTHDATE': 'BIRTHDATE'
})
df_Bank

DEBT  OVERDUE_DAYS  INITIAL_LIMIT   BIRTHDATE      SEX  \
0      6063.50             3         7000.0  1983-07-08  Мужской   
1      3765.04             0         5000.0  1987-12-19  Женский   
2      2067.66             0         2650.0  1966-02-28  Женский   
3      2370.39             0         3000.0  1972-02-18  Женский   
4      2280.55             0         3000.0  1997-02-23  Женский   
...        ...           ...            ...         ...      ...   
18415     0.00             0        10850.0  1999-11-08  Мужской   
18416     0.00             0        13000.0  1993-06-03  Мужской   
18417     0.00             0         3500.0  2002-11-05  Мужской   
18418     0.00             0        10000.0  2003-09-15  Мужской   
18419     0.00             0         1500.0  1994-01-14  Мужской   

                       EDU   INCOME  TERM CREDIT_HISTORY_RATING  \
0      Среднее специальное   703.07    60                    A1   
1                   Высшее  1693.68    60                    B2   
2                  Среднее   724.49    60                    C2   
3      Среднее специальное  1045.84    60                    A3   
4                   Высшее  1092.65    60                    B3   
...                    ...      ...   ...                   ...   
18415               Высшее  3563.75    60                    B1   
18416  Среднее специальное  2865.72    36                   NaN   
18417  Среднее специальное  3363.05    24                    C2   
18418              Среднее  1670.13    60                    C1   
18419               Высшее  3517.90    60                    C1   

              LIVING_AREA SETTLEMENT_NAME               INDUSTRY_NAME   PDN  \
0      Гомельская область       КОСАКОВКА                         АПК  0.98   
1                     NaN           МИНСК       Финансы и страхование  0.17   
2             ГРОДНЕНСКАЯ            ЁДКИ  Медицина и здравоохранение  0.40   
3                 МИНСКАЯ        ФАНИПОЛЬ  Государственное управление  0.29   
4                     NaN           Минск  Медицина и здравоохранение  0.63   
...                   ...             ...                         ...   ...   
18415            г. Минск           МИНСК   Информационные технологии  0.40   
18416     Минская область            Жалы               Строительство  0.19   
18417     Минская область       Солигорск               Строительство  0.19   
18418            г. Минск           МИНСК                    Торговля  0.28   
18419   Брестская область          КОБРИН                  МВД/МЧС/МО  0.26   

       CLIENT_ID  SCORING_MARK  UNDERAGE_CHILDREN_COUNT  VELCOM_SCORING  \
0         919517         233.0                        0             NaN   
1         539353           NaN                        1             NaN   
2         818913           NaN                        0             NaN   
3         691598         216.0                        0             NaN   
4         425193           NaN                        0             NaN   
...          ...           ...                      ...             ...   
18415     687867         344.0                        0            -2.0   
18416     984627         108.0                        2            -2.0   
18417    1060373         129.0                        0            -2.0   
18418    1056128         179.0                        0            -2.0   
18419    1061453         269.0                        0            -2.0   

       FAMILY_STATUS  
0                  1  
1                  1  
2                  2  
3                  2  
4                  2  
...              ...  
18415              1  
18416              1  
18417              2  
18418              2  
18419              2  

[18418 rows x 18 columns]

In [4]:
df_Bank.shape

(18420, 18)

In [5]:
df_Bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18420 entries, 0 to 18419
Data columns (total 18 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   DEBT                     18420 non-null  float64
 1   OVERDUE_DAYS             18420 non-null  int64  
 2   INITIAL_LIMIT            18420 non-null  float64
 3   BIRTHDATE                18420 non-null  object 
 4   SEX                      18420 non-null  object 
 5   EDU                      18420 non-null  object 
 6   INCOME                   18420 non-null  float64
 7   TERM                     18420 non-null  int64  
 8   CREDIT_HISTORY_RATING    17488 non-null  object 
 9   LIVING_AREA              16297 non-null  object 
 10  SETTLEMENT_NAME          18374 non-null  object 
 11  INDUSTRY_NAME            18420 non-null  object 
 12  PDN                      18420 non-null  float64
 13  CLIENT_ID                18420 non-null  int64  
 14  SCORING_MARK          

In [6]:
# 
df_Bank.describe()

DEBT  OVERDUE_DAYS  INITIAL_LIMIT        INCOME          TERM  \
count  18420.000000  18420.000000   18420.000000  18420.000000  18420.000000   
mean    4318.769750      9.196743    5127.834256   1664.840782     52.301737   
std     4336.432353     42.309092    4879.201691   1048.813447     13.378788   
min        0.000000      0.000000     180.000000      0.000000     23.000000   
25%     1287.900000      0.000000    1550.000000   1021.692500     36.000000   
50%     2675.670000      0.000000    3200.000000   1460.175000     60.000000   
75%     5785.870000      0.000000    6912.500000   1980.460000     60.000000   
max    40000.000000    491.000000   40000.000000  29942.820000     60.000000   

                PDN     CLIENT_ID  SCORING_MARK  UNDERAGE_CHILDREN_COUNT  \
count  18420.000000  1.842000e+04  16786.000000             18420.000000   
mean       0.290288  8.226245e+05    186.432622                 0.636754   
std        0.144359  1.966853e+05     67.430848                 0.854167   
min        0.000000  1.289800e+04   -108.000000                 0.000000   
25%        0.190000  7.193478e+05    134.000000                 0.000000   
50%        0.300000  9.083645e+05    182.000000                 0.000000   
75%        0.390000  9.621360e+05    234.000000                 1.000000   
max        1.210000  1.061453e+06    416.000000                 3.000000   

       VELCOM_SCORING  FAMILY_STATUS  
count     8889.000000   18420.000000  
mean       177.040162       1.498751  
std        310.624940       0.500988  
min         -2.000000       1.000000  
25%         -2.000000       1.000000  
50%         -2.000000       1.000000  
75%        278.000000       2.000000  
max        998.000000       4.000000

In [7]:
df_Bank.isnull().sum()

DEBT                          0
OVERDUE_DAYS                  0
INITIAL_LIMIT                 0
BIRTHDATE                     0
SEX                           0
EDU                           0
INCOME                        0
TERM                          0
CREDIT_HISTORY_RATING       932
LIVING_AREA                2123
SETTLEMENT_NAME              46
INDUSTRY_NAME                 0
PDN                           0
CLIENT_ID                     0
SCORING_MARK               1634
UNDERAGE_CHILDREN_COUNT       0
VELCOM_SCORING             9531
FAMILY_STATUS                 0
dtype: int64

There are missing values in: 

CREDIT_HISTORY_RATING =932(object)

LIVING_AREA = 2123(object)

SETTLEMENT_NAME = 46(object)

SCORING_MARK = 1634(float64)

VELCOM_SCORING  = 9531(float64)

In [8]:
# Not sure what to do with missing values
df_Bank


DEBT  OVERDUE_DAYS  INITIAL_LIMIT   BIRTHDATE      SEX  \
0      6063.50             3         7000.0  1983-07-08  Мужской   
1      3765.04             0         5000.0  1987-12-19  Женский   
2      2067.66             0         2650.0  1966-02-28  Женский   
3      2370.39             0         3000.0  1972-02-18  Женский   
4      2280.55             0         3000.0  1997-02-23  Женский   
...        ...           ...            ...         ...      ...   
18415     0.00             0        10850.0  1999-11-08  Мужской   
18416     0.00             0        13000.0  1993-06-03  Мужской   
18417     0.00             0         3500.0  2002-11-05  Мужской   
18418     0.00             0        10000.0  2003-09-15  Мужской   
18419     0.00             0         1500.0  1994-01-14  Мужской   

                       EDU   INCOME  TERM CREDIT_HISTORY_RATING  \
0      Среднее специальное   703.07    60                    A1   
1                   Высшее  1693.68    60                    B2   
2                  Среднее   724.49    60                    C2   
3      Среднее специальное  1045.84    60                    A3   
4                   Высшее  1092.65    60                    B3   
...                    ...      ...   ...                   ...   
18415               Высшее  3563.75    60                    B1   
18416  Среднее специальное  2865.72    36                   NaN   
18417  Среднее специальное  3363.05    24                    C2   
18418              Среднее  1670.13    60                    C1   
18419               Высшее  3517.90    60                    C1   

              LIVING_AREA SETTLEMENT_NAME               INDUSTRY_NAME   PDN  \
0      Гомельская область       КОСАКОВКА                         АПК  0.98   
1                     NaN           МИНСК       Финансы и страхование  0.17   
2             ГРОДНЕНСКАЯ            ЁДКИ  Медицина и здравоохранение  0.40   
3                 МИНСКАЯ        ФАНИПОЛЬ  Государственное управление  0.29   
4                     NaN           Минск  Медицина и здравоохранение  0.63   
...                   ...             ...                         ...   ...   
18415            г. Минск           МИНСК   Информационные технологии  0.40   
18416     Минская область            Жалы               Строительство  0.19   
18417     Минская область       Солигорск               Строительство  0.19   
18418            г. Минск           МИНСК                    Торговля  0.28   
18419   Брестская область          КОБРИН                  МВД/МЧС/МО  0.26   

       CLIENT_ID  SCORING_MARK  UNDERAGE_CHILDREN_COUNT  VELCOM_SCORING  \
0         919517         233.0                        0             NaN   
1         539353           NaN                        1             NaN   
2         818913           NaN                        0             NaN   
3         691598         216.0                        0             NaN   
4         425193           NaN                        0             NaN   
...          ...           ...                      ...             ...   
18415     687867         344.0                        0            -2.0   
18416     984627         108.0                        2            -2.0   
18417    1060373         129.0                        0            -2.0   
18418    1056128         179.0                        0            -2.0   
18419    1061453         269.0                        0            -2.0   

       FAMILY_STATUS  
0                  1  
1                  1  
2                  2  
3                  2  
4                  2  
...              ...  
18415              1  
18416              1  
18417              2  
18418              2  
18419              2  

[18420 rows x 18 columns]

In [9]:
# Checking for duplicates
df_Bank.duplicated().sum()

2

In [10]:
# Drop duplicates
df_Bank.drop_duplicates(inplace=True)

# Check if duplicates have been dropped
df_Bank.duplicated().sum()

0

In [11]:
print(df_Bank['CREDIT_HISTORY_RATING'].value_counts())

B3    2154
C1    2079
C2    2011
C3    2006
B2    1701
D1    1387
B1    1335
D2    1060
A3     996
D3     869
A1     777
A2     640
E1     200
E3     144
E2     123
-1       4
Name: CREDIT_HISTORY_RATING, dtype: int64


## **2.DATA CLEANING**

## **3.EDA (DATA ANALYSIS) & DATA VISUALIZATION**